File: src/main/java/com/lucas/matches/infrastructure/repository/MatchRepository.java
package com.lucas.matches.infrastructure.repository;

import com.lucas.matches.infrastructure.document.MatchDocument;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface MatchRepository extends MongoRepository<MatchDocument, String> {
    List<MatchDocument> findByMatchIdIn(List<String> matchIds);
    List<MatchDocument> findByTournamentIdIn(List<String> tournamentIds);
}

File: src/main/kotlin/com/lucas/matches/infrastructure/repository/MatchRepository.kt
package com.lucas.matches.infrastructure.repository

import com.lucas.matches.infrastructure.document.MatchDocument
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface MatchRepository : MongoRepository<MatchDocument, String> {
    fun findByMatchIdIn(matchIds: List<String>): List<MatchDocument>
    fun findByTournamentIdIn(tournamentIds: List<String>): List<MatchDocument>
}